applications/nrf_desktop/src/modules/led_stream.c

```diff
@@ -0,0 +1,205 @@
+            last_added_position = -1;
+        }
+        last_added_position++;
+        LOG_INF("Insert data on position %u, free places %u", last_added_position, free_places);
+        queue_data(data, last_added_position);
+        free_places--;
+    } else {
+        LOG_INF("Queue is full - dropping incoming step");
+    }
+}
+
+static void send_data_from_queue()
+{
+    if (free_places < QUEUE_SIZE) {
+        LOG_INF("Sending data %u, %u, %u from position %u, free places %u", steps_queue[send_position].color.c[0],
+                steps_queue[send_position].color.c[1], steps_queue[send_position].color.c[2], send_position, free_places);
+
+        led_stream_effect.steps = &steps_queue[send_position];
+        send_effect(&led_stream_effect);
+
+        send_position++;
+        if (send_position == QUEUE_SIZE) {
+            send_position = 0;
+        }
+        free_places++;
+    } else {
+        LOG_INF("No steps ready in queue");
+        last_added_position = -1;
+        send_position = 0;
+        first_sent = false;
+    }
+}
+
+static void handle_incoming_step(const struct config_event *event)
+{
+    if (leds_initialized) {
+        store_data(&event->dyndata);
+        if (!first_sent) {
+            LOG_INF("Sending first led effect");
+            send_data_from_queue();
+            first_sent = true;
+        }
+    } else {
+        LOG_WRN("Leds module not ready");
+    }
+}
+
+static void handle_led_ready_event(const struct led_ready_event *event)
+{
+    LOG_INF("Led ready event handler");
+    send_data_from_queue();
+}
+
+static void handle_config_event(const struct config_event *event)
+{
+    if (!event->store_needed) {
+        /* Accept only events coming from transport. */
+        return;
+    }
+
+    if (GROUP_FIELD_GET(event->id) != EVENT_GROUP_LED_STREAM) {
+        /* Only LED STREAM events. */
+        return;
+    }
+
+    switch (TYPE_FIELD_GET(event->id)) {
+        case STREAM_DATA:
+            handle_incoming_step(event);
+            break;
+
+        default:
+            /* Ignore unknown event. */
+            LOG_WRN("Unknown LED STREAM config event");
+            break;
+    }
+}
+
+static void handle_config_fetch_request_event(const struct config_fetch_request_event *event)
+{
+    LOG_INF("Handle config fetch request event");
+
+    if (GROUP_FIELD_GET(event->id) != EVENT_GROUP_LED_STREAM) {
+        /* Only LED STREAM events. */
+        return;
+    }
+
+    LOG_INF("Free places left: %u", free_places);
+
+    size_t data_size = sizeof(free_places);
+    struct config_fetch_event *fetch_event =
+            new_config_fetch_event(data_size);
+
+    fetch_event->id = event->id;
+    fetch_event->recipient = event->recipient;
+    fetch_event->channel_id = event->channel_id;
+
+    fetch_event->dyndata.data[0] = free_places;
+
+    EVENT_SUBMIT(fetch_event);
+}
+
+static bool event_handler(const struct event_header *eh)
+{
+    if (IS_ENABLED(CONFIG_DESKTOP_CONFIG_CHANNEL_ENABLE)) {
```

Comment:
WE may think about some sort of depends on here (no way to use led_stream without config channel)

Reply:
done